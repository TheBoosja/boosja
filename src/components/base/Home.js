import React from 'react';
import styled from 'styled-components';
import { Heading, Anchor } from '../styled';

const Layout = styled.div`
	padding: 2rem 5rem;
`;

const Home = () => {
	return (
		<Layout>
			<Heading>Boosja</Heading>
			<Anchor to='/search'>To the Search</Anchor>
		</Layout>
	);
};

export default Home;