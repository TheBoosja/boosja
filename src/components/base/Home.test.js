import React from 'react';
import { shallow } from 'enzyme';
import Home from './Home';

describe('Home', () => {
	const home = shallow(<Home />);

	it('renders properly', () => {
		expect(home).toMatchSnapshot();
	});

	it('contains a Heading component', () => {
		expect(home.find('Heading').exists()).toBe(true);
	});

	it('contains a link to the Search page', () => {
		expect(home.find('Anchor').props().to).toEqual('/search');
	});
});