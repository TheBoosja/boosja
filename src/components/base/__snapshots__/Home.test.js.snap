// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Heading>
          Boosja
</Heading>,
        <Anchor
          to="/search"
>
          To the Search
</Anchor>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Boosja",
        },
        "ref": null,
        "rendered": "Boosja",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "To the Search",
          "to": "/search",
        },
        "ref": null,
        "rendered": "To the Search",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Heading>
            Boosja
</Heading>,
          <Anchor
            to="/search"
>
            To the Search
</Anchor>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Boosja",
          },
          "ref": null,
          "rendered": "Boosja",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "To the Search",
            "to": "/search",
          },
          "ref": null,
          "rendered": "To the Search",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
